testkit/tests/runtime.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::{
+    blockchain::InstanceConfig,
+    crypto::{PublicKey, SecretKey},
+    exonum_merkledb::{Fork, Snapshot},
+    node::ApiSender,
+    proto::Any,
+    runtime::dispatcher::{Dispatcher, DispatcherSender},
+    runtime::{
+        ArtifactId, ArtifactProtobufSpec, CallInfo, ExecutionContext, ExecutionError,
+        InstanceDescriptor, InstanceSpec, Runtime, StateHashAggregator,
+    },
+};
+use exonum_testkit::{runtime::RuntimeFactory, TestKitBuilder};
+use futures::{Future, IntoFuture};
+use std::{cell::RefCell, sync::Arc};
+
+// Tracks parts of state of runtime that we're interested in.
+#[derive(Debug)]
+struct RuntimeState {
+    deployed_artifact: ArtifactId,
+    deploy_spec: Any,
+    config_params: Any,
+}
+
+// Main purpose is to track and make some assertions on state of runtime.
+#[derive(Debug)]
+struct RuntimeTester {
+    state: RefCell<RuntimeState>,
```

Comment:
I you use `RwLock` here you can avoid unsafe impl.

Reply:
Done.